docs/linting/Typed_Linting.mdx

```diff
@@ -3,9 +3,41 @@ id: typed-linting
 title: Linting with Type Information
 ---
 
+import Tabs from '@theme/Tabs';
+import TabItem from '@theme/TabItem';
+
 Some typescript-eslint rules utilize the awesome power of TypeScript's type checking APIs to provide much deeper insights into your code.
 To tap into TypeScript's additional powers, there are two small changes you need to make to your config file:
 
+<Tabs groupId="eslint-config">
+<TabItem value="Flat Config">
+
+```js title="eslint.config.js"
+export default tseslint.config(
+  eslint.configs.recommended,
+  // Remove this line
+  ...tseslint.configs.recommended,
+  // Added lines start
+  ...tseslint.configs.recommendedTypeChecked,
+  languageOptions: {
+    parserOptions: {
+      project: true,
+      tsconfigRootDir: import.meta.dirname,
```

Comment:
This was only added in node v21
https://nodejs.org/api/esm.html#importmetadirname

It's probably too early to add this considering that a lot of people will still be on CJS and few people use the odd-number releases.

Reply:
Heh, and that's why I hadn't added it in originally. I'm not a fan of showing using the big verbose `dirname(fileURLToPath(import.meta.url))` 😬. 

Nit: it's [also in Node 20.11.0](https://nodejs.org/en/blog/release/v20.11.0).

What do you suggest we do instead? Leave this out? Have the verbose thing?

My preference would be to leave it in and have it just be `undefined` for older users.

Reply:
Oh it finally got backported? We support 18 as well and I'm sure people haven't upgraded.

I think we can add a quick aside to the first example maybe and leave it at that then?
The alternative is CJS - cos I don't want to polyfil it with the ugly url hack.